components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java

```diff
@@ -467,6 +494,44 @@ private void deleteServiceProvider(String applicationName,
         }
     }
 
+    /**
+     * Delete OAuth application when there is no associated service provider exists.
+     *
+     * @param appDTO       {@link OAuthConsumerAppDTO} object of the OAuth app to be deleted
+     * @param tenantDomain Tenant Domain
+     * @param username     User Name
+     * @throws DCRMException
+     */
+    private void deleteOAuthApplicationWithoutAssociatedSP(OAuthConsumerAppDTO appDTO, String tenantDomain,
+                                                           String username) throws DCRMException {
+
+        ApplicationManagementService applicationManagementService = DCRDataHolder.getInstance()
+                .getApplicationManagementService();
+        try {
+            if (log.isDebugEnabled()) {
+                log.debug("Delete OAuth application with the consumer key: " + appDTO.getOauthConsumerKey());
+            }
+            applicationManagementService.deleteOAuthApplication(appDTO.getOauthConsumerKey());
```

Comment:
Why we need to invoke through ApplicationManagementService? Can't we directly use oauth service?

Reply:
fixed